ui/src/components/Views/Configs/MountPointForm.tsx

```diff
@@ -1,137 +1,187 @@
-      setMountPointUser('tmp');
+  const getMountPointPath = (): string => {
+    if(ddClient.host.platform === 'darwin'){
+      return `/Users/${userState.selectedUser || 'loading...'}/Library/Caches/localstack/volume`;
     }
-    const foundUsers = res.stdout.split('\n');
-    foundUsers.pop(); // remove last '' element
-    setUserState({ loading: false, selectedUser: foundUsers[0], users: foundUsers });
+    return `/home/${userState.selectedUser || 'loading...'}/.cache/localstack/volume`;
   };
 
-  useEffect(() => {
-    const execChecks = async () => {
-      if (userState.users.length === 0) {
+  const checkWindowsDistro = async () => {
+    setOsState({ ...osState, loading: true });
 
-        setHasLocalImage({ ...hasLocalImage, checking: true });
+    const res = await ddClient.extension.host?.cli.exec('checkWSLOS.cmd', []);
+    
+    const foundOSs = getOSsFromBinary(res.stdout);
+
+    setOsState({ loading: false, selectedOS: foundOSs[0], OSs: foundOSs });
+    setTriggerUserCheck(!triggerUserCheck);
+  };
+
+  const checkUser = async () => {
+    setUserState({ ...userState, loading: true });
+
+    let res: ExecResult;
+    let foundUsers = [];
+    if (ddClient.host.platform === 'win32') {
+      res = await ddClient.extension.host?.cli.exec('checkUser.cmd', [osState.selectedOS]);
+      foundUsers = getUsersFromBinaryWindows(res.stdout);
+    } else {
+      res = await ddClient.extension.host?.cli.exec('checkUser.sh', []);
+      foundUsers = getUsersFromBinaryUnix(res.stdout);
+    }
+
+    if (res.stderr || !res.stdout) {
+      ddClient.desktopUI.toast.error(`Error while locating users: ${res.stderr} using /tmp as mount point`);
+      setUserState({ loading: false, selectedUser: '../tmp', users: ['tmp'] });
+      setMountPointData('../tmp');
+    }
 
-        const images = await ddClient.docker.listImages() as [DockerImage];
-        const isPresent = images.some(image => image.RepoTags?.at(0) === LATEST_IMAGE);
+    setUserState({ loading: false, selectedUser: foundUsers[0], users: foundUsers });
+  };
 
-        setHasLocalImage({ checking: false, isPresent });
+  const locateMountPoint = async () => {
+    if (ddClient.host.platform === 'win32') {
+      checkWindowsDistro();
+    } else {
+      checkUser();
+    }
+  };
 
-        if (isPresent) {
-          checkHomeDir();
-        } else {
-          setIsPullingImage(true);
-        }
+  useEffect(() => {
+    const execChecks = async () => {
+      if (userState.users.length === 0
+        || (ddClient.host.platform === 'win32' && osState.OSs.length === 0)) {
+        locateMountPoint();
       }
     };
 
     execChecks();
-  }, [triggerImageCheck]);
+  }, []);
+
+  useEffect(() => {
+    if (osState.selectedOS) {
+      checkUser();
+    }
+  }, [triggerUserCheck]);
 
   const onClose = () => {
-    setMountPointUser(userState.selectedUser);
+    setMountPointData(`${userState.selectedUser},${osState.selectedOS}`);
```

Comment:
Wondering if there could be a more elegant way to pass the user and OS. Rather than a comma-separated string which we're later splitting, could we just use an array/tuple?